src/EditorFeatures/Test/CommentSelection/ToggleLineCommentCommandHandlerTests.cs

```diff
@@ -0,0 +1,1084 @@
+        {
+            var markup =
+@"
+class C
+{
+    void M()
+    {
+        [|//var i = 1;
+        var j = 2;|]
+    }
+}";
+            var expected =
+@"
+class C
+{
+    void M()
+    {
+[|        ////var i = 1;
+        //var j = 2;|]
+    }
+}";
+
+            ToggleComment(markup, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.ToggleLineComment)]
+        public void AddComment_MultipleLinesWithCommentsInLineSelected()
+        {
+            var markup =
+@"
+class C
+{
+    void M()
+    {
+        [|var i = 1; // A comment.
+        var j = 2;|]
+    }
+}";
+            var expected =
+@"
+class C
+{
+    void M()
+    {
+[|        //var i = 1; // A comment.
+        //var j = 2;|]
+    }
+}";
+
+            ToggleComment(markup, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.ToggleLineComment)]
+        public void AddComment_MultipleLinesWithDifferentIndentationsSelected()
+        {
+            var markup =
+@"
+class C
+{
+[|    void M()
+    {
+        var i = 1;
+
+        var j = 2;
+    }|]
+}";
+            var expected =
+@"
+class C
+{
+[|    //void M()
+    //{
+    //    var i = 1;
+
+    //    var j = 2;
+    //}|]
+}";
+
+            ToggleComment(markup, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.ToggleLineComment)]
+        public void AddComment_MultiCaret()
+        {
+            var markup =
+@"
+class C
+{
+    void M()
+    {
+        var [||]i = 1;
+        var [||]j = 2;
+    }
+}";
+            var expected =
+@"
+class C
+{
+    void M()
+    {
+[|        //var i = 1;|]
+[|        //var j = 2;|]
+    }
+}";
+
+            ToggleComment(markup, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.ToggleLineComment)]
+        public void AddComment_MultiSeletion()
+        {
+            var markup =
+@"
+class C
+{
+    void M()
+    {
+        [|var i = 1;|]
+        [|var j = 2;|]
+    }
+}";
+            var expected =
+@"
+class C
+{
+    void M()
+    {
+[|        //var i = 1;|]
+[|        //var j = 2;|]
+    }
+}";
+
+            ToggleComment(markup, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.ToggleLineComment)]
+        public void AddComment_MultiSeletionPartiallyCommented()
+        {
+            var markup =
+@"
+class C
+{
+    void M()
+    {
+        [|//var i = 1;|]
+        [|var j = 2;|]
+    }
+}";
+            var expected =
+@"
+class C
+{
+    void M()
+    {
+[|        ////var i = 1;|]
+[|        //var j = 2;|]
+    }
+}";
+
+            ToggleComment(markup, expected);
+        }
+
+        [WpfFact, Trait(Traits.Feature, Traits.Features.ToggleLineComment)]
+        public void AddComment_VisualBasicStyleComment()
+        {
+            var markup =
+@"
+Class A
+    [|Function M()
+        Dim a = 1
+
+    End Function|]
+End Class";
+            var expected =
+@"
+Class A
+[|    'Function M()
```

Comment:
I think in real editor, after this, auto vb auto formatter run right?

Reply:
as above, not sure the question - it will format on uncomment (and the formatting might look off here b/c of the [| markers in the test string)